packages/next/client/image.tsx

```diff
@@ -133,6 +140,7 @@ function generateImgAttrs({
   width,
   quality,
   sizes,
+  loader,
```

Comment:
Just as an idea: in my concept PR, I [used a default parameter here](https://github.com/vercel/next.js/pull/20213/files#diff-ed7e3de801c4bc992b2796b6dd476f27320efef5d2911101e2ccb7b27c35866aR188) and it simplified control flow a bit removed the need for `callLoader`.

Follow the code path there and let me know if you like it!

Reply:
I adopted this pattern, as part of removing the CallLoader pattern as suggested. I also added a little additional logic to handle the passing of strings to specify other default loaders, as @style suggested above.